pkg/disttask/framework/proto/priority_queue.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package proto
+
+// TaskWrapper wraps a task and adds some extra information for the priority queue.
+type TaskWrapper struct {
+	*Task
+
+	IndexInPriorityQueue int // Index of the task in the priority queue
+}
+
+// TaskPriorityQueue represents a priority queue of tasks.
+type TaskPriorityQueue []*TaskWrapper
+
+// TaskPriorityQueue implementation for heap.Interface
+func (pq TaskPriorityQueue) Len() int { return len(pq) }
+
+// Less returns true if the task at index i has higher priority than the task at index j.
+func (pq TaskPriorityQueue) Less(i, j int) bool {
```

Comment:
using this after it merged https://github.com/pingcap/tidb/pull/49195/files#diff-9cbb4aa7310f95ed10a52e14e392efa29e0356b179e85efec38db8fde5dca9f3R165